BuildingBlocks/src/DevelopmentKit.Identity/ValueObjects/IdentityAddress.cs

```diff
@@ -45,37 +47,38 @@ public static bool IsValid(string? stringValue)
 
         var lengthIsValid = stringValue.Length <= MAX_LENGTH;
 
-        var realm = stringValue[..3];
+        const string pattern = @"^did\:e\:((?:[a-zA-Z]+\.)+[a-zA-Z]+)\:dids\:(.+)(.{2})$";
+
+        var matches = Regex.Matches(stringValue, pattern, RegexOptions.IgnoreCase).First().Groups;
```

Comment:
`matches` is not a good name for something that contains the `Groups` of the first match.